apps/server/lib/lexical/server/boot.ex

```diff
@@ -78,8 +79,8 @@ defmodule Lexical.Server.Boot do
   defp load_app_modules(app_name) do
     Application.ensure_loaded(app_name)
 
-    with {:ok, modules} <- :application.get_key(app_name, :modules) do
-      Enum.each(modules, &Code.ensure_loaded!/1)
+    with {:ok, modules} <- Application.spec(app_name, :modules) do
+      Code.ensure_all_loaded!(modules)
```

Comment:
This function was introduced in elixir 1.15

Reply:
yes, but the we already `alias Future.Code`

and the logic inside is simple, so I think it is fine.

```elixir
  @spec ensure_all_loaded!([module]) :: :ok
  def ensure_all_loaded!(modules) do
    case ensure_all_loaded(modules) do
      :ok ->
        :ok

      {:error, errors} ->
        formatted_errors =
          errors
          |> Enum.sort()
          |> Enum.map_join("\n", fn {module, reason} ->
            "  * #{inspect(module)} due to reason #{inspect(reason)}"
          end)

        raise ArgumentError, "could not load the following modules:\n\n" <> formatted_errors
    end
  end
  ```

Reply:
The erlang call that it uses was introduced in erlang 19, so it's compatible.